File: cp_rust/src/bin/abc354-b-at-coder-janken-2.rs
// Created by Ayush Biswas at 2026/07/11 18:23
// https://atcoder.jp/contests/adt_easy_20240724_2/tasks/abc354_b
use cp_lib::*;

// @code begin
use cpio::*;
use itertools::Itertools;

sol! {
    fn solution(
        n: usize,
        players: [[String; 2]; n]
    ) -> String {
        let mut names = vec![];
        let mut total_score: usize = 0;
        for [p, s] in players {
            names.push(p);
            total_score += s.parse::<usize>().unwrap();
        }
        names
            .into_iter()
            .sorted()
            .nth(total_score % n)
            .unwrap()
    }
}
// @code end

File: cp_nim/src/bin/abc354-b-at-coder-janken-2.nim
# Created by Ayush Biswas at 2026/07/19 11:00
# https://atcoder.jp/contests/adt_easy_20240724_2/tasks/abc354_b
import ../lib/cpio

# @code begin

import strutils, algorithm

var r = newReader()

let
  n = r[int]
  a = r[seq[seq[string]], n].sortedByIt(it[0])

var totalScore: int
for user in a:
  totalScore += user[1].parseInt

echo a[totalScore mod n][0]

# @code end
